loadtest/control/actions.go

```diff
@@ -486,3 +486,145 @@ func ViewUser(u user.User) UserActionResponse {
 	}
 	return UserActionResponse{Info: fmt.Sprintf("viewed user %s", member.UserId)}
 }
+
+// Reload simulates the given user reloading the page
+// while connected to the server by executing the API
+// calls seen during a real page reload.
+func Reload(u user.User) UserActionResponse {
+	// Getting preferences.
+	err := u.GetPreferences()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	prefs, _ := u.Store().Preferences()
+	var userIds []string
+	chanId := ""
+	for _, p := range prefs {
+		switch {
+		case p.Name == model.PREFERENCE_NAME_LAST_CHANNEL:
+			chanId = p.Value
+		case p.Category == model.PREFERENCE_CATEGORY_DIRECT_CHANNEL_SHOW:
+			userIds = append(userIds, p.Name)
+		}
+	}
+
+	if chanId != "" {
+		// Marking the channel as viewed
+		_, err := u.ViewChannel(&model.ChannelView{
+			ChannelId:     chanId,
+			PrevChannelId: "",
+		})
+		if err != nil {
+			return UserActionResponse{Err: NewUserError(err)}
+		}
+	}
+
+	if ok, err := u.IsSysAdmin(); ok && err != nil {
+		err = u.GetConfig()
+		if err != nil {
+			return UserActionResponse{Err: NewUserError(err)}
+		}
+	}
+
+	err = u.GetClientLicense()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	// Getting the user.
+	userId, err := u.GetMe()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	roles, _ := u.Store().Roles()
+	var roleNames []string
+	for _, role := range roles {
+		roleNames = append(roleNames, role.Name)
+	}
+	if len(roleNames) > 0 {
+		_, err = u.GetRolesByNames(roleNames)
+		if err != nil {
+			return UserActionResponse{Err: NewUserError(err)}
+		}
+	}
+
+	err = u.GetWebappPlugins()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	_, err = u.GetAllTeams(0, 50)
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	_, err = u.GetTeamsForUser(userId)
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	team, err := u.Store().RandomTeamJoined()
```

Comment:
When I reload the page I shouldn't join to a random team :)

Reply:
This is not joining a team but picking one to view. What's happening on the webapp is that we store the last team we've seen somewhere where it survives reloads/closing the browser. Since we are not doing it here we need to pick a team that we will use as the first team to load.

Reply:
I failed on expressing the idea by saying "join", I should've say "pick" :) Still, we shouldn't load a random team after reload. We should be on the last team that we were in.

Reply:
Yes, this is a known limitation and I suggest we just add a comment documenting this fact and move on for now because properly solving this would require keeping state out of memory (file, db).

I think the most important thing here is that we make sure an API call gets made to fetch a team during reload.  
The fact that the team fetched was the last seen might not have such an impact overall.  
 Worst case we are generating a few more cache misses on the server but that's yet to be measured. I suggest we get back to this kind of details once we have run several load-tests.

Reply:
Agreed, thanks for the clarification 👍